mime/multipart: allow custom content types to be given for form files

robtimus opened the pull request:
The existing implementation always uses application/octet-stream as content-type for form files.
This change allows developers to give a custom content type instead

robtimus commented:
CLA has been signed.

sneko commented:
Hi @robtimus , any idea why this is not merged? Looks like a required thing.

In my case an external API requires me to specify the right MIME type... so having `application/octet-stream` just breaks the request 😞 

Thank you,

robtimus commented:
Not a clue why this still isn't merged. I even took care of keeping the API intact by adding a new function for this.

ianlancetaylor commented:
In Go new API is expected to go through the proposal process (https://golang.org/s/proposal), as Emmanuel pointed out on https://golang.org/cl/153178.

robtimus commented:
@ianlancetaylor: I missed the update that specified that I had to create a proposal. I just created one, https://github.com/golang/go/issues/46771, and added a link to this PR. I've also rebased the fork for this PR from the latest master branch so it's up-to-date again.
